### ISSUE TITLE
[iOS9] Non-Renewing subscription

### ISSUE THREAD

**Issue by chancezeus** [j3k0/cordova-plugin-purchase#314]
I've been trying to get a non-renewing subscription working (with thanks to the iOS review team that did not approve of the correctly working and syncing version based on consumables). But I don't seem to be able to get it to work...

I have programmed a verification/subscription server that checks the iOS receipt and returns the results (using the codes in the documentation). Everything seems to work up until the point where the product needs to be expired and/or repurchased, the only thing that happens is the message "This in-app purchase has already been bought. It will be restored for free."

It does not matter if I (manually) finish the product or anything.

Whilst debugging through the code I noticed that in the storekitFinish function a product.transactions field is used, but in my product this field is undefined. Is this an error with the plugin or should the field be returned/filled by the server somehow?

**Comment by headshota** [comment]
Just curious, how did you register the non-renewing subscription. The store object does not have a constant for this type of products. 

https://github.com/j3k0/cordova-plugin-purchase/blob/master/src/js/constants.js#L11
```
/*///*/     store.FREE_SUBSCRIPTION = "free subscription";
/*///*/     store.PAID_SUBSCRIPTION = "paid subscription";
/*///*/     store.CONSUMABLE        = "consumable";
/*///*/     store.NON_CONSUMABLE    = "non consumable";
```

These are the only ones available.

**Comment by chancezeus** [comment]
I gave it the type store.PAID_SUBSCRIPTION (maybe I was wrong assuming 
this would cover both RENEWING and NON-RENEWING subscriptions)...

**Comment by nod** [comment]
I, too, am wrestling with non-renewable subscriptions.  Any luck with this?

**Comment by headshota** [comment]
After trying out the plugin for a while this is what my findings are so far:

`store.PAID_SUBSCRIPTION`  indeed covers the non-renewing subscriptions. So if you set up everything correctly in Itunes Store and in the app, you should be able to perform a purchase.

"This in-app purchase has already been bought. It will be restored for free." message I only get when during validation (when verify is called). the callback returns error, but without `store.PURCHASE_EXPIRED` code. If the code is set correctly, it successfully expires the subscription.

NOTE: there is an error in the docs regarding how the code need to be set in the callback. Here's a correct form:

```
callback(false, {
  code: store.PURCHASE_EXPIRED,
  error: {
    message: "Subscription Expired"
  }
});
```

This should be called in the `store.validator` function in order to expire the subscription.

**Comment by anonrig** [comment]
I am having the exact same problem, any suggestions?

**Comment by bradrlaw** [comment]
I am having this problem on Android.  On iOS it is working as expected and the subscription expires / allows a repurchase.  My validator function is returning the expired code as shown above.

**Comment by seibelj** [comment]
https://github.com/j3k0/cordova-plugin-purchase/pull/412

**Comment by j3k0** [comment]
Good news everyone!

Some improvements have been made in non-renewing subscription support (newly released version 6.0). I also created a [helper library for apps that just need non-renewing subscriptions](https://github.com/j3k0/cordova-non-renewing-subscription). Feel free to take a look, to use it or see how it's implemented for reference.
